### PR TITLE
Styled Components: Fix interpolated className

### PR DESCRIPTION
**Pull request by ItsJonQ**
This update consolidates how the interpolated className is created, in an attempt to resolve https://github.com/WordPress/gutenberg/pull/29084#issuecomment-788372734

Changes:

* A new `getInterpolatedClassName` util is created and exported from `@wp-g2/create-styles`
* `getInterpolatedClassName` ensures the `@emotion/hash` function is only called if the argument is valid
* `@wp-g2/context` adds `@wp-g2/create-styles` as a dependency (was previously missing)
* All calls to `@emotion/hash` go through the new `getInterpolatedClassName` util

### CODE REVIEW

**Comment by ItsJonQ** [comment]
Going to try an alpha `0.0.160` release to test in Gutenberg

File: packages/create-styles/src/create-style-system/utils.js [hunk 1]
```diff
@@ -167,3 +168,14 @@ export function compileInterpolatedStyles(interpolatedStyles, props) {
...
+
+/**
+ * Creates an interpolated className. This is used when styled components are
+ * interpolated in the template literal.
+ *
+ * @param {string} displayName The displayName of the component.
+ * @returns {string} The interpolated className.
+ */
+export function getInterpolatedClassName(displayName) {
+	return typeof displayName === 'string' ? `ic-${hash(displayName)}` : '';
```
Comment (sarayourfriend) [inline packages/create-styles/src/create-style-system/utils.js:180]: Should we raise a development-env warning that the displayName was undefined?

**Review by sarayourfriend — APPROVED** [review]

File: packages/create-styles/src/create-style-system/utils.js (hunk 1 above)
Comment (ItsJonQ) [reply packages/create-styles/src/create-style-system/utils.js:180]: Good idea. But maybe not here. Perhaps in `contextConnect()` or `useContext()`?

What you think :)

Comment (sarayourfriend) [reply packages/create-styles/src/create-style-system/utils.js:180]: Yup, that makes sense to me!

Comment (ItsJonQ) [reply packages/create-styles/src/create-style-system/utils.js:180]: I just pushed that update :). Lemme know if that works

File: packages/context/src/context-connect.js [hunk 2]
```diff
@@ -35,6 +35,12 @@ export function contextConnect(Component, namespace, options = {}) {
 		? namespace[0]
 		: namespace || WrappedComponent.name;
 
+	if (process.env.NODE_ENV === 'development') {
+		if (typeof namespace === 'undefined') {
+			console.warn('contextConnect', 'Please provide a namespace.');
+		}
+	}
+
```
Comment (sarayourfriend) [inline packages/context/src/context-connect.js:43]: We can use `@wordpress/warning` for this 🙂

Comment (ItsJonQ) [reply packages/context/src/context-connect.js:43]: We could, but I'd prefer not adding another `@wordpress/` dependency for now 🙈 haha

**Pull request merged**